.github/workflows/build-debian.msan-based.yml

```diff
@@ -28,23 +28,16 @@ jobs:
       fail-fast: false
       matrix:
         include:
-          - image: debian:11
+          - image: debian:12
```

Comment:
Would this and the change to `debian.Dockerfile` mean that we’d lose MSAN coverage for the MariaDB Server 10.6 and 10.11 branches? I don’t care that much about the soon-to-EOL 10.5.

Reply:
Nope still there. We won't generate packaging for them but they still build and test.

For 10.5:
```
buildbot@7b53fedf2e3b:/build$ mysql-test/mtr --mem --big-test --force --retry=0 --max-test-fail=100 --parallel=16

Completed: All 5247 tests were successful.

961 tests were skipped, 305 by the test itself.
```